src/pages/media/AttachmentModalScreen/AttachmentModalContext.tsx

```diff
@@ -0,0 +1,56 @@
+import React, {useCallback, useEffect, useMemo, useRef} from 'react';
+import useCurrentReportID from '@hooks/useCurrentReportID';
+import type ChildrenProps from '@src/types/utils/ChildrenProps';
+import type {AttachmentModalScreenParams} from './types';
+
+type AttachmentModalContextValue = {
+    isAttachmentHidden: (reportActionID: string) => boolean;
+    updateHiddenAttachments: (reportActionID: string, isHidden: boolean) => void;
+    setCurrentAttachment: (attachmentProps: AttachmentModalScreenParams | undefined) => void;
+    getCurrentAttachment: () => AttachmentModalScreenParams | undefined;
+};
+
+const AttachmentModalContext = React.createContext<AttachmentModalContextValue>({
+    isAttachmentHidden: () => false,
+    updateHiddenAttachments: () => {},
+    setCurrentAttachment: () => {},
+    getCurrentAttachment: () => undefined,
+});
+
+function AttachmentModalContextProvider({children}: ChildrenProps) {
+    const currentReportID = useCurrentReportID();
```

Comment:
`const {currentReportID} = useCurrentReportID();` May look less confusing later in the array of dependencies. But's its not a strong opinion

Reply:
`currentReportID` can be null, so this wouldn't work. I would have to have a fallback object, which then gets reinitialized every render.

Like this: `const {currentReportID} = useCurrentReportID() ?? {};`

I think i prefer this approach, which should cause the least re-renders. Also this is just copied over from `AttachmentModal` and used the same way across the codebase, so i'd say it's out of scope for this PR